app/services/FinancialDetailsService.scala

```diff
@@ -60,64 +61,58 @@ class FinancialDetailsService @Inject()(val financialDetailsConnector: Financial
-        case financialDetails: FinancialDetailsModel => Some(financialDetails)
-        case error: FinancialDetailsErrorModel if error.code != NOT_FOUND => Some(error)
-        case _ => None
-      }
-    }
-  }
-
-  @deprecated("Use getUnpaidFinancialDetailsV2 instead", "MISUV-8845")
-  def getAllUnpaidFinancialDetails()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[List[FinancialDetailsResponseModel]] = {
-    getAllFinancialDetails.map { chargesWithYears =>
-      chargesWithYears.flatMap {
-        case (_, errorModel: FinancialDetailsErrorModel) => Some(errorModel)
-        case (_, financialDetails: FinancialDetailsModel) =>
-          val unpaidDocDetails: List[DocumentDetail] = financialDetails.unpaidDocumentDetails()
-          if (unpaidDocDetails.nonEmpty) Some(financialDetails.copy(documentDetails = unpaidDocDetails)) else None
-      }
+      val maxYears = 5
+      val listOfCalls = yearsOfMigration.reverse.grouped(maxYears).toList
+
+      Future.sequence(listOfCalls.map { years =>
+        val (from, to) = (years.min, years.max)
+        Logger("application").debug(s"Getting financial details for TaxYears: ${from} - ${to}")
+
+        for {
+          response <- financialDetailsConnector.getFinancialDetails(TaxYear.forYearEnd(from), TaxYear.forYearEnd(to), user.nino)
+        } yield response match {
+          case financialDetails: FinancialDetailsModel => Some(financialDetails)
+          case error: FinancialDetailsErrorModel if error.code != NOT_FOUND => Some(error)
+          case _ => None
+        }
+      }).map(x => {
+        x.foldLeft[Option[FinancialDetailsResponseModel]](None) { (acc, next) =>
+          combineTwoResponses(acc, next)
+        }
+      })
     }
   }
 
-  def getAllUnpaidFinancialDetailsV2()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[FinancialDetailsResponseModel]] = {
-    getAllFinancialDetailsV2.map {
+  def getAllUnpaidFinancialDetails()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[FinancialDetailsResponseModel]] = {
+    getAllFinancialDetails.map {
       case Some(errorModel: FinancialDetailsErrorModel) => Some(errorModel)
       case Some(financialDetails: FinancialDetailsModel) =>
         val unpaidDocDetails: List[DocumentDetail] = financialDetails.unpaidDocumentDetails()
         if (unpaidDocDetails.nonEmpty) Some(financialDetails.copy(documentDetails = unpaidDocDetails)) else None
       case _ => None
     }
   }
+
+  def combineTwoResponses(response1: Option[FinancialDetailsResponseModel],
+                                  response2: Option[FinancialDetailsResponseModel]): Option[FinancialDetailsResponseModel] = {
+    (response1, response2) match {
+      case (Some(validModel1: FinancialDetailsModel), Some(validModel2: FinancialDetailsModel)) => Some(validModel1.mergeLists(validModel2))
+      case (Some(validModel: FinancialDetailsModel), _) => Some(validModel)
```

Comment:
its looks like based on this change we can merge valid and not valid/error model?

Reply:
If we get multiple calls back and one of them is an error/not valid, we should just take the valid one. I think if for example:
We request for 2020-2025 and 2017-2019. 2020-2025 returns data but 2017-2019 returns 404. 
I think we should just take the 2020-2025 response and we don't need to worry about the 404

Reply:
if these are agree business ruleS specified in the story, I am ok with that